### PR TITLE
.Net core 3 support

### PR DESCRIPTION
**Pull request by ProH4Ck**
This let it works with a .net core 3 project and solves:

`System.TypeLoadException: 'Could not load type 'Microsoft.AspNetCore.Mvc.MvcJsonOptions' from assembly 'Microsoft.AspNetCore.Mvc.Formatters.Json, Version=3.0.0.0, Culture=neutral, PublicKeyToken=adb9793829ddae60'.'
`

### CODE REVIEW

**Comment by ProH4Ck** [comment]
@rasmus I think build is blocked by some variables set on your AppVeyor account (relative to SQL Server). 
The same build is working fine in my AppVeyor account: https://ci.appveyor.com/project/ProH4Ck/eventflow/builds/28040687

**Comment by i-dentify** [comment]
Any updates on this? I am soooo soooo waiting for this merge

**Comment by ProH4Ck** [comment]
> Any updates on this? I am soooo soooo waiting for this merge

I can't do anything... Build is working fine on my appveyor account. @rasmus we need you!

File: Source/EventFlow.AspNetCore/Configuration/EventFlowJsonOptionsMvcConfiguration.cs [hunk 1]
```diff
@@ -27,7 +27,13 @@
 
 namespace EventFlow.AspNetCore.Configuration
 {
-    public class EventFlowJsonOptionsMvcConfiguration : IConfigureOptions<MvcJsonOptions>
+    public class EventFlowJsonOptionsMvcConfiguration : 
+#if NETSTANDARD2_0
+        IConfigureOptions<MvcJsonOptions>
+#endif
+#if NETCOREAPP3_0
+        IConfigureOptions<MvcNewtonsoftJsonOptions>
```
Comment (leotsarev) [inline Source/EventFlow.AspNetCore/Configuration/EventFlowJsonOptionsMvcConfiguration.cs]: I think that using ASP.NET Core 2.1 on .NET Core is possible. This scenario will fail, isn't it?

**Comment by rasmus** [comment]
Hi @ProH4Ck thanks for the PR, but it seems like the integration tests for MSSQL and EF are broken though.

```
A network-related or instance-specific error occurred while establishing a connection to SQL Server. The server was not found or was not accessible. Verify that the instance name is correct and that SQL Server is configured to allow remote connection
```

👍 for moving the integration test setup to the official images!

**Comment by ProH4Ck** [comment]
> Hi @ProH4Ck thanks for the PR, but it seems like the integration tests for MSSQL and EF are broken though.
> 
> ```
> A network-related or instance-specific error occurred while establishing a connection to SQL Server. The server was not found or was not accessible. Verify that the instance name is correct and that SQL Server is configured to allow remote connection
> ```
> 
> 👍 for moving the integration test setup to the official images!

Hi @rasmus 
they're failing because probably you have ms sql connection string hard coded in your appveyor account.
The same build is working fine on my appveyor: 
https://ci.appveyor.com/project/ProH4Ck/eventflow/builds/28040687

I have no variables set in my account:

image

I tried to move sql server to docker but memory limits on appveyor prevent container to start...

**Comment by Brandon2255p** [comment]
Dotnet core 2.2 is reaching its end of life by the end of this year. 
Could @rasmus please outline what is outstanding? I am going to work to push this PR over the line :)

**Comment by rasmus** [comment]
@Brandon2255p I'm going through the settings and everything seems to be as should be.

My wife has started evening school which means I don't have as much time for spare time projects as I would like, as I have the kids full time. I do try to keep up, but I'll try to be more available.

**Comment by rasmus** [comment]
I'll see if I can do a separate update of the image to 2019 tonight, removing the need for it in this PR.

**Comment by rasmus** [comment]
I'll try to get the image change working in #709

**Comment by rasmus** [comment]
Found the problem in #709, will hopefully get it merged tonight. If it does, I'll merge develop into this PR as well to see if fixes the build here as well.

**Comment by i-dentify** [comment]
> Found the problem in #709, will hopefully get it merged tonight. If it does, I'll merge develop into this PR as well to see if fixes the build here as well.

Hooray!

**Comment by Brandon2255p** [comment]
Amazing! Thanks so much. Really appreciate that you made the time

**Comment by rasmus** [comment]
@Brandon2255p just keep poking me if I don't respond. Its my responsibility and I need to make the time.

There's a problem with the license in several of the source files (not from this PR). Basically the header isn't correct in accordance to the CLA. I'm correcting this in #712 and will release as soon as its done.

I try to keep the legal stuff correct, but darn its hard.

**Comment by ProH4Ck** [comment]
Finally merged! Thanks!
Maybe it's time for 1.0.0 release?

**Comment by promontis** [comment]
@rasmus awesome! Thanks for the great work! Is there already a new NuGet release for this?

**Comment by rasmus** [comment]
Yes, as soon as I get out of bed, had to leave work due to sickness. I hope to do it within the next few hours.

**Comment by promontis** [comment]
@rasmus take your time :) no need to rush

**Comment by rasmus** [comment]
@promontis managed to get out of bed and start the release. You can follow it here

https://ci.appveyor.com/project/eventflow/eventflow/builds/29444767

I'll check on it in a few hours to see if everything is ok. The GitHub release draft with artifacts needs to be published manually, but the NuGet packages will be uploaded during the build. As soon as NuGet then indexes the packages, you should be able to upgrade.

**Comment by promontis** [comment]
It's up \o/ \o/ \o/

**Pull request merged**